libnmstate/nm/sriov.py

```diff
@@ -43,11 +45,88 @@ def create_setting(iface_state, base_con_profile):
         if not sriov_setting:
             sriov_setting = nmclient.NM.SettingSriov.new()
 
+        if sriov_config.get(Ethernet.SRIOV.VFS_SUBTREE):
+            _create_sriov_vfs_in_setting(
+                copy.deepcopy(sriov_config[Ethernet.SRIOV.VFS_SUBTREE]),
+                sriov_setting,
+            )
+
         sriov_setting.props.total_vfs = sriov_config[Ethernet.SRIOV.TOTAL_VFS]
 
     return sriov_setting
 
 
+def _create_sriov_vfs_in_setting(vfs_config, sriov_setting):
+    vf_id_object_list = {vf.get_index(): vf for vf in sriov_setting.props.vfs}
+    for vf in vfs_config:
+        vf_id = vf[Ethernet.SRIOV.VFS.ID]
+        if vf_id in vf_id_object_list.keys():
+            vf_object = vf_id_object_list[vf_id].dup()
+        else:
+            vf_object = nmclient.NM.SriovVF.new(vf[Ethernet.SRIOV.VFS.ID])
+        if vf.get(Ethernet.SRIOV.VFS.MAC):
+            vf_object.set_attribute(
+                nmclient.NM.SRIOV_VF_ATTRIBUTE_MAC,
+                nmclient.GLib.Variant.new_string(vf[Ethernet.SRIOV.VFS.MAC]),
+            )
+        if vf.get(Ethernet.SRIOV.VFS.TRUST):
+            vf_object.set_attribute(
+                nmclient.NM.SRIOV_VF_ATTRIBUTE_TRUST,
+                nmclient.GLib.Variant.new_boolean(
+                    vf[Ethernet.SRIOV.VFS.TRUST]
+                ),
+            )
+        if vf.get(Ethernet.SRIOV.VFS.MAX_TX_RATE):
+            vf_object.set_attribute(
+                nmclient.NM.SRIOV_VF_ATTRIBUTE_MAX_TX_RATE,
+                nmclient.GLib.Variant.new_uint32(
+                    vf[Ethernet.SRIOV.VFS.MAX_TX_RATE]
+                ),
+            )
+        if vf.get(Ethernet.SRIOV.VFS.MIN_TX_RATE):
+            vf_object.set_attribute(
+                nmclient.NM.SRIOV_VF_ATTRIBUTE_MIN_TX_RATE,
+                nmclient.GLib.Variant.new_uint32(
+                    vf[Ethernet.SRIOV.VFS.MIN_TX_RATE]
+                ),
+            )
+        if vf.get(Ethernet.SRIOV.VFS.SPOOF_CHECK):
+            vf_object.set_attribute(
+                nmclient.NM.SRIOV_VF_ATTRIBUTE_SPOOF_CHECK,
+                nmclient.GLib.Variant.new_boolean(
+                    vf[Ethernet.SRIOV.VFS.SPOOF_CHECK]
+                ),
+            )
+        if vf.get(Ethernet.SRIOV.VFS.VLANS_SUBTREE):
+            _create_vf_vlan_setting(
+                vf[Ethernet.SRIOV.VFS.VLANS_SUBTREE], vf_object
+            )
+        sriov_setting.add_vf(vf_object)
+
+
+def _create_vf_vlan_setting(vlan_config, vf_object):
+    for vlan in vlan_config:
+        vlan_id = vlan_config[Ethernet.SRIOV.VFS.VLANS.ID]
+        vf_object.add_vlan(vlan_id)
+        if vlan_config.get(Ethernet.SRIOV.VFS.VLANS.PRIORITY):
+            vf_object.set_vlan_qos(
+                vlan_id, vlan_config[Ethernet.SRIOV.VFS.VLANS.PRIORITY]
+            )
+        if vlan_config.get(Ethernet.SRIOV.VFS.VLANS.PROTOCOL):
+            # pylint: disable=no-member
```

Comment:
Why this line? Is that caused by `nmclient.NM.SriovVFVlanProtocol._1AD`?

Reply:
Not needed anymore. Thanks!